engine/src/test/java/io/camunda/zeebe/engine/processing/timer/DueDateTimerCheckerTest.java

```diff
@@ -18,22 +18,62 @@
 import static org.mockito.Mockito.when;
 
 import io.camunda.zeebe.engine.api.TaskResultBuilder;
+import io.camunda.zeebe.engine.processing.scheduled.DueDateTriggerState;
 import io.camunda.zeebe.engine.processing.timer.DueDateTimerChecker.TriggerTimersSideEffect;
 import io.camunda.zeebe.engine.processing.timer.DueDateTimerChecker.YieldingDecorator;
-import io.camunda.zeebe.engine.state.immutable.TimerInstanceState;
 import io.camunda.zeebe.engine.state.immutable.TimerInstanceState.TimerVisitor;
 import io.camunda.zeebe.engine.state.instance.TimerInstance;
 import io.camunda.zeebe.protocol.record.intent.TimerIntent;
-import io.camunda.zeebe.scheduler.clock.ActorClock;
 import java.time.Duration;
-import java.util.function.Consumer;
+import org.agrona.concurrent.EpochClock;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Nested;
 import org.junit.jupiter.api.Test;
 import org.mockito.Mockito;
 
 class DueDateTimerCheckerTest {
 
+  private static final class PinnedClock implements EpochClock {
```

Comment:
Didn't we have a controllable actor clock which you can pin  🤔

Reply:
Yes, but imo we don't need a whole actor clock here.